src/overlays/actors/ovl_En_Owl/z_en_owl.c

```diff
@@ -40,137 +71,1139 @@ const ActorInit En_Owl_InitVars = {
+            }
+            break;
+
+        case ENOWL_GET_TYPE_3:
+            if (CHECK_QUEST_ITEM(15)) {
+                Actor_MarkForDeath(&this->actor);
+                return;
+            }
+            break;
+    }
+
+    this->unk_3DA = 0;
+    this->unk_3F0 = this->actor.home.pos.y;
+    this->unk_3EC = this->actor.home.rot.y;
+    func_8095A510(this, globalCtx);
+
+    switch (owlType) {
+        case ENOWL_GET_TYPE_1:
+            this->actionFunc = func_8095AEC0;
+            break;
+
+        case ENOWL_GET_TYPE_2:
+            this->actionFunc = func_8095BE0C;
+            if (gSaveContext.weekEventReg[9] & 0x20) {
+                this->actor.textId = 0xBF0;
+            } else {
+                this->actor.textId = 0xBEA;
+            }
+            break;
+
+        case ENOWL_GET_TYPE_3:
+            this->actionFunc = func_8095AFEC;
+            break;
+
+        case ENOWL_GET_TYPE_1000:
+            this->actionFunc = func_8095BF20;
+            break;
+
+        default:
+            this->actionFlags |= 2;
+            this->unk_3DA = 0x20;
+            this->actionFunc = func_8095BF58;
+            break;
+    }
+}
+
+void EnOwl_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnOwl* this = THIS;
+
+    if (ENOWL_GET_TYPE(&this->actor) != ENOWL_GET_TYPE_30) {
+        Collider_DestroyCylinder(globalCtx, &this->collider);
+    }
+}
+
+void func_8095A920(EnOwl* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    this->actor.world.rot.y = Math_Vec3f_Yaw(&this->actor.world.pos, &player->actor.world.pos);
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.world.rot.y, 6, 0x3E8, 0xC8);
+}
+
+s32 func_8095A978(EnOwl* this, GlobalContext* globalCtx, u16 textId, f32 targetDist, f32 arg4) {
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        return true;
+    }
+
+    this->actor.textId = textId;
+    if (this->actor.xzDistToPlayer < targetDist) {
+        this->actor.flags |= 0x10000;
+        func_800B8500(&this->actor, globalCtx, targetDist, arg4, 0);
+    }
+
+    return false;
+}
+
+s32 func_8095A9FC(EnOwl* this, GlobalContext* globalCtx, u16 textId) {
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        return true;
+    }
+
+    this->actor.textId = textId;
+    if (this->actor.xzDistToPlayer < 120.0f) {
+        func_800B8500(&this->actor, globalCtx, 350.0f, 1000.0f, 0);
+    }
+
+    return false;
+}
+
+void func_8095AA70(EnOwl* this) {
+    EnOwl_ChangeMode(this, func_8095C1C8, func_8095C484, &this->skelAnime1, &D_0600CB94, 0.0f);
+    this->eyeTexIndex = 0;
+    this->blinkTimer = Rand_S16Offset(60, 60);
+}
+
+void func_8095AAD0(EnOwl* this, GlobalContext* globalCtx) {
+    s32 switchFlag = ENOWL_GET_SWITCHFLAG(&this->actor);
+
+    if (switchFlag < 0x7F) {
+        Actor_SetSwitchFlag(globalCtx, switchFlag);
+    }
+
+    func_8095AA70(this);
+}
+
+void func_8095AB1C(EnOwl* this, GlobalContext* globalCtx) {
+    if (!(this->unk_3DA & 0x3F)) {
```

Comment:
```suggestion
    if ((this->unk_3DA % 64) == 0) {
```
I think something like this would be more appropriate if it matches. Applies to a few spots in this code

Reply:
This works in most of them, but not the one in the if.